src/Chapter3.hs

```diff
@@ -907,9 +977,39 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+{- ORMOLU_ENABLE -}
 class Append a where
-    append :: a -> a -> a
+  append :: a -> a -> a
+
+newtype Gold = Gold Int
 
+instance Append Gold where
+  append (Gold x) (Gold y) = Gold $ x + y
+
+-- copy pasta from previous section
+data List a
+  = Empty
+  | Cons a (List a)
+
+instance Append (List a) where
+  append x y = combine Empty x y
+    where
+      -- rebuild cons, unlike implmenting reverse in previous chapter
+      teardown :: List a -> List a -> List a
+      teardown acc Empty = acc
+      teardown acc (Cons z zs) = Cons z (teardown acc zs)
+      -- reconstruct the list, one after another
+      combine :: List a -> List a -> List a -> List a
+      combine acc Empty Empty = acc
+      combine acc xs ys = combine (teardown (teardown acc xs) ys) Empty Empty
+
+-- note2self: constraint means type a implements Append already, and now we are implementing for (Maybe a)
+instance (Append a) => Append (Maybe a) where
+  append _ Nothing = Nothing
+  append Nothing _ = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality, it would be more beneficial is this function returned `Just` when at least one argument is `Just`.